# Probability Math

In [3]:
p_coffee_drinker = .65
p_cancer = 0.005
p_coffee_drinker_given_cancer = .85

p_cancer_drinker_coffee_drinker = (p_coffee_drinker_given_cancer 
                                   * p_cancer) / p_coffee_drinker

p_cancer_drinker_coffee_drinker

0.006538461538461539

P (A|B) = P (B|A)P(A) / P (B)

In [4]:
p = .50**10
p

0.0009765625

#### Binomial distribution

In [4]:
from scipy.stats import binom
n = 10
p = 0.5

for k in range(n+1):
    probability = binom.pmf(k,n,p)
    print('{0}-{1}'.format(k, probability))

0-0.0009765625
1-0.009765625000000002
2-0.04394531250000004
3-0.1171875
4-0.2050781249999999
5-0.24609375000000003
6-0.2050781249999999
7-0.11718749999999999
8-0.04394531250000004
9-0.009765625000000002
10-0.0009765625


#### Beta distribution

In [8]:
from scipy.stats import beta

a = 8
b = 2

p = beta.cdf(.9,a,b)
p

0.7748409780000002

According to our calculation, ther is 77.48% chance the underlaying probability of success
is 90% or less.

In [12]:
# shows how we calculate this subtraction operation, for new test

a = 30
b = 6

p = 1 - beta.cdf(.9,a,b)
p

0.13163577484183697

So this means out of 8/10 successful engine test, ter is only a 22.5% chance the underlying
success rate is 90%.

In [14]:
from scipy.stats import beta

a=8
b=2

p = beta.cdf(.90,a,b) - beta.cdf(.80,a,b)
p

0.33863336199999994

## Exercises

1. There is a 30% chance of rain today, and a 40% chance your umbrella order
will arrive on time. You are eager to walk in the rain today and cannot do so
without either! What is the probability it will rain and your umbrella will arrive?

In [15]:
rain = .3
umbrella = .4
rain_with_umbrella = rain * umbrella
rain_with_umbrella

# there is 12% of chances that 

0.12

2. What is the probability will not rain or the umbrella arives?

In [37]:
not_rain = .7
umbrella = .4

not_run = (not_rain + umbrella) - (not_rain * umbrella)
not_run

# The outcome yielded says that there is 82% of chance that we finish the day dry. 

0.8200000000000001

3. The expected delivery on time decrease with rain until 20%, so what is the probability it will raing and your umbrella will arrive on time.

In [18]:
rain = .3
umbrella_with_rain = .2

umbrella_on_time = rain * umbrella_with_rain
umbrella_on_time

0.06

4. 137 passenger booked on a flight from las vegas to dallas. However it is Las vegas on sunday morning and you estimate each passenger is% likely to not show up. You're trying to figure out how many seats to overbook so the plane does not fly empty. 
How likely is it at least 50 passenger will not show up

In [31]:
n = 137
p = 0.4

for k in range(50+138):
    probability += binom.pmf(k,n,p)
    print('{0} - {1}'.format(k, probability))

0 - 4.043163486926879e-31
1 - 3.733187619595884e-29
2 - 1.7113812559386846e-27
3 - 5.193286264822027e-26
4 - 1.1735459470758458e-24
5 - 2.1063484644259044e-23
6 - 3.1278258553627447e-22
7 - 3.952325653808102e-21
8 - 4.338070889341985e-20
9 - 4.20140815405262e-19
10 - 3.635160390972988e-18
11 - 2.838106863928248e-17
12 - 2.0160242637744651e-16
13 - 1.3119957452118304e-15
14 - 7.86860391356731e-15
15 - 4.371139523391054e-14
16 - 2.259122511123217e-13
17 - 1.0904731750843816e-12
18 - 4.9329661705158304e-12
19 - 2.097705973038723e-11
20 - 8.408382773254813e-11
21 - 3.184803945977178e-10
22 - 1.142419841760137e-09
23 - 3.888884665634863e-09
24 - 1.2586023274571507e-08
25 - 3.879340094950043e-08
26 - 1.1405561375955343e-07
27 - 3.20329826646364e-07
28 - 8.605718127784847e-07
29 - 2.2142816171095555e-06
30 - 5.463185147504134e-06
31 - 1.2939156712068014e-05
32 - 2.9448593917146593e-05
33 - 6.446861223094951e-05
34 - 0.00013588198291007654
35 - 0.0002759882149091263
36 - 0.0005406333197962202


5. You flipped a coin 19 times and got heads 15 and tails 4 times.
Do you think this con has any good probability of being fair? Why or ahy not?

In [51]:
head = 15
tail = 4

p = beta.cdf(.50,head,tail)
p
#there is only a 0.037% o chance, so for me it's a tricky coin

0.0037689208984375